src/lib/rules/http-cache/http-cache.ts

```diff
@@ -0,0 +1,431 @@
+                    parsed.usedDirectives.set(directive, null);
+                } else {
+                    parsed.invalidDirectives.set(directive, null);
+                }
+
+                return parsed;
+            },
+            {
+                header: cacheControlHeader,
+                invalidDirectives: new Map(),
+                invalidValues: new Map(),
+                usedDirectives: new Map<string, number>()
+            });
+
+            return parsedCacheControlHeader;
+        };
+
+        /**
+         * Transforms a Map of directives to a string with the following format:
+         *
+         * directive1=value1
+         * directive2
+         * diretive3=value3
+         */
+        const directivesToString = (directives: Map<string, any>) => {
+            let str = '';
+
+            directives.forEach((val, key) => {
+
+                if (str.length > 0) {
+                    str += '\n';
+                }
+
+                str += key;
+                if (val) {
+                    str += `=${val}`;
+                }
+            });
+
+            return str;
+        };
+
+        /**
+         * Compares if the `max-age` or `s-maxage` directives of `directives` are smaller (<0),
+         * equal (0), or bigger (>0) than the given `threshold`.
+         */
+        const compareToMaxAge = (directives: Directives, threshold: number): number => {
+            const maxAge = directives.get('max-age');
+            const sMaxAge = directives.get('s-maxage');
+
+            if (maxAge) {
+                return maxAge === threshold ? 0 : maxAge - threshold;
+            }
+
+            if (sMaxAge) {
+                return sMaxAge === threshold ? 0 : sMaxAge - threshold;
+            }
+
+            return -1;
+        };
+
+        /*
+         * ------------------------------------------------------------------------------
+         * Directive validators
+         * ------------------------------------------------------------------------------
+         */
+
+        /**
+         * Prevents agains the usage of non recommended directives (`must-revalidate`)
+         */
+        const nonRecommendedDirectives = (directives: Directives): string => {
+            const noDirectives = ['must-revalidate', 'no-store'];
+
+            for (const noDirective of noDirectives) {
+                if (directives.has(noDirective)) {
+                    return noDirective;
+                }
+            }
+
+            return null;
+        };
+
+        /*
+         * Validate if cache-control exists, not having this header is
+         * an error because it's up to the browser vendor to decide what
+         * to do.
+         */
+        const hasCacheControl = async (directives: ParsedDirectives, fetchEnd: IFetchEnd): Promise<boolean> => {
+            const { resource, response: { headers } } = fetchEnd;
+            const cacheControl: string = headers && headers['cache-control'] || null;
```

Comment:
Don't know if it's possible, but if the value of the cache-control header is " ", the following check will pass.

Use: `normalizeString(headers && headers['cache-control'])`?